server_lib.js

```diff
@@ -129,51 +137,98 @@ exports.callbackTemplates = {
 
     "notAuthorized": function (request, response) {
         console.log("403 error: " + request.url);
-        response.writeHead(404, {"Content-Type": "text/html"});
+        response.writeHead(403, {"Content-Type": "text/html"});
         response.write("403 Not Found\n");
         response.end();
     }
 };
 
-exports.getStaticServer = function(_urlCallbackDict) {
-    let urlCallbackDict = _urlCallbackDict || {};
+class Router {
+    constructor() {
+        this._urlDict = {};
+    }
+
+    addPlainURL(url, bind) {
+        this._urlDict[url] = bind;
+    }
+
+    addRegexURL(url, bind) {
+        this._urlDict[getRegexStr(url)] = bind;
+    }
+
+    addPrefixURL(url, bind) {
+        this._urlDict[getPrefixStr(url)] = bind;
+    }
+
+    get getURLDict() {
+        return this._urlDict;
+    }
+}
+
+function getStaticServer (_URLMap) {
+    let urlCallbackDict = _URLMap.getURLDict;
 
-    return exports.http.createServer(function (request, response) {
+    return http.createServer(function (request, response) {
         let callback = urlCallbackDict[request.url] || null;
 
-        if (!callback) {
+        if (callback) {
+            callback(request, response);
+            console.log("Complete match succeeded: " + request.url);
+            return
+        } else {
             console.log("Complete match failed: " + request.url);
-            for (let url in urlCallbackDict) {
-                if (isRegExpStr(url) && request.url.match(new RegExp(regexFromRegexStr(url)))) {
-                    callback = urlCallbackDict[url];
-                    console.log("URL " + request.url +  " match regex: " + url);
-                    break;
-                }
-            }
+        }
 
-            if (!callback) {
-                console.log("Regex match failed: " + request.url);
-            } else {
-                callback(request, response);
-                return
+        for (let url in urlCallbackDict) {
+            if (_isRegExpStr(url) && request.url.match(new RegExp(_regexFromRegexStr(url)))) {
+                callback = urlCallbackDict[url];
+                console.log("URL " + request.url +  " match regex: " + url);
+                break;
             }
+        }
 
-            for (let url in urlCallbackDict) {
-                if (isPrefixStr(url) && request.url.startsWith(prefixFromPrefixStr(url))) {
-                    callback = urlCallbackDict[url];
-                    console.log("URL " + request.url + " match prefix " + url);
+        if (callback) {
+            callback(request, response);
+            return
+        } else {
+            console.log("Regex match failed: " + request.url);
+        }
+
+        let urlPair = {
```

Comment:
Напиши что-ли комментарий, чтобы было более очевидно, зачем это.
Например
```javascript
/*
    For fix some interesting case:
    Dirs:    
        dir_1/mid_dir/file, 
        dir_2/file
    Routing: 
        /static/ => dir_1,  
        /static/mid_dir => dir_2
    Url:     
        /static/mid_dir/file

    Return dir_1/mid_dir/file (instead of dir_2/mid_dir/file) 
    if you do not check the length of the prefix!
*/
```